### PR TITLE
Make the officer delta handler generic for better code reuse

### PR DESCRIPTION
**Pull request by amoriartyCH**
We found a way to turn the OfficerDeltaHandler into a generic DeltaHandler
by passing in during the route register stage of application bootup
a topic choice per route. This means that the DeltaHandler can be reused
by all new delta routes.

Also removed use of duplicated parameters in CHValidator by using it's New constructor to store the api spec location instead of passing it around.

Updated mocks using mock-gen

fixed unit tests where failing due to new logic

Go fmt ./...

**Resolves DSND-135**

### CODE REVIEW

File: docs/adding-a-new-delta.md [hunk 1]
```diff
@@ -27,77 +25,12 @@ paths:
...
-    w.WriteHeader(http.StatusOK)
-}
 ```
 
-## 4. Exposing your new delta endpoint
+## 3. Exposing your new delta endpoint
 The final step of adding a new delta is to register your new route within the `register.go` file.
 ```go
-appRouter.HandleFunc("/delta/example-delta", NewExampleDeltaHandler(kSvc, h, chv, cfg).ServeHTTP).Methods(http.MethodPost).Name("example-delta")
+appRouter.HandleFunc("/delta/example-delta", NewDeltaHandler(kSvc, h, chv, cfg, true, cfg.ExampleDeltaTopic).ServeHTTP).Methods(http.MethodPost).Name("example-delta")
```
Comment (mreis-ch) [inline docs/adding-a-new-delta.md:33]: Here, might be worth explaining they can add a validate endpoint as well and what the boolean flag is for

File: validation/chOpenApiValidator.go [hunk 2]
```diff
@@ -54,14 +55,15 @@ func NewCHValidator(openApiSpec string) (CHValidator, error) {
...
+		doc:         doc,
+		openApiSpec: openApiSpec,
 	}, nil
 }
 
-// ValidateRequestAgainstOpenApiSpec takes a request and an openAPI spec location (string relative path) and uses the
-// spec to validate the provided request. If any validation errors are found, then they are formatted and returned to the
-// caller. If any errors are encountered while attempting to validate, they are handled and also returned to the caller.
-func (chv CHValidatorImpl) ValidateRequestAgainstOpenApiSpec(httpReq *http.Request, openApiSpec, contextId string) ([]byte, error) {
+// ValidateRequestAgainstOpenApiSpec takes a request and uses an openAPI3 spec to validate the it. If any validation errors
```
Comment (mreis-ch) [inline validation/chOpenApiValidator.go:63]: Maybe expand on the "to validate the it"! :D Previously it was "to validate the provided request" but maybe put request body

File: docs/adding-a-new-delta.md (hunk 1 above)
Comment (amoriartyCH) [reply docs/adding-a-new-delta.md]: Added this, good point!

File: validation/chOpenApiValidator.go (hunk 2 above)
Comment (amoriartyCH) [reply validation/chOpenApiValidator.go]: Changed to mention the request's body as it's more descriptive

**Review by mreis-ch — APPROVED** [review]

**Pull request merged**